src/lib/pubkey/ecc_key/ecc_key.h

```diff
@@ -12,9 +12,13 @@
 
 #include <botan/ec_group.h>
 #include <botan/pk_keys.h>
+#include <memory>
 
 namespace Botan {
 
+class EC_PublicKey_Data;
+class EC_PrivateKey_Data;
+
 /**
 * This class represents abstract ECC public keys. When encoding a key
 * via an encoder that can be accessed via the corresponding member
```

Comment:
Done for both

Reply:
GCC points out a valid issue

```
2024-07-23T23:04:58.5003263Z build/include/public/botan/ecc_key.h:143:30: error: defaulted move assignment for ‘Botan::EC_PrivateKey’ calls a non-trivial move assignment operator for virtual base ‘Botan::EC_PublicKey’ [-Werror=virtual-move-assign]
2024-07-23T23:04:58.5005697Z   143 | class BOTAN_PUBLIC_API(2, 0) EC_PrivateKey : public virtual EC_PublicKey,
2024-07-23T23:04:58.5006842Z       |                              ^~~~~~~~~~~~~
```

```
 ‘-Wno-virtual-move-assign’
     Suppress warnings about inheriting from a virtual base with a
     non-trivial C++11 move assignment operator.  This is dangerous
     because if the virtual base is reachable along more than one path,
     it is moved multiple times, which can mean both objects end up in
     the moved-from state.  If the move assignment operator is written
     to avoid moving from a moved-from object, this warning can be
     disabled.
```

I truly cannot wait for the absolute slog of decoupling public and private keys in Botan4.

For now I've marked the move constructor/operator= as `delete`